p2p/src/protocol/cbfmgr/cache.rs

```diff
@@ -164,6 +165,97 @@ impl FilterCache {
         None
     }
 
+    /// Returns the range of cached filter heights that intersects with the provided
+    /// range.
+    ///
+    /// Returns [`None`] if there is no overlap.
+    ///
+    /// ```
+    /// use nakamoto_p2p::protocol::cbfmgr::cache::FilterCache;
+    /// use nakamoto_common::block::filter::BlockFilter;
+    ///
+    /// let mut cache = FilterCache::new(32);
+    ///
+    /// cache.push(3, BlockFilter::new(&[]));
+    /// cache.push(4, BlockFilter::new(&[]));
+    /// cache.push(5, BlockFilter::new(&[]));
+    ///
+    /// assert_eq!(cache.intersection(1..=4), Some(3..=4));
+    /// assert_eq!(cache.intersection(5..=5), Some(5..=5));
+    /// assert_eq!(cache.intersection(3..=3), Some(3..=3));
+    /// assert_eq!(cache.intersection(1..=8), Some(3..=5));
+    /// assert_eq!(cache.intersection(4..=7), Some(4..=5));
+    ///
+    /// assert_eq!(cache.intersection(9..=11), None);
+    /// assert_eq!(cache.intersection(1..=2), None);
+    ///
+    /// ```
```

Comment:
Are these code blocks run as part of 'cargo test' or similar?

Reply:
yes!